finality-aleph/src/validator_network/manager/mod.rs

```diff
@@ -223,16 +208,15 @@ impl<A: Data, D: Data> Manager<A, D> {
     /// Remove a peer from the list of peers that we want to stay connected with.
     /// Close any incoming and outgoing connections that were established.
     pub fn remove_peer(&mut self, peer_id: &AuthorityId) {
-        self.addresses.remove(peer_id);
-        self.incoming.remove(peer_id);
-        self.outgoing.remove(peer_id);
+        self.wanted.remove_peer(peer_id);
+        self.have.remove(peer_id);
     }
 
     /// Send data to a peer.
     /// Returns error if there is no outgoing connection to the peer,
     /// or if the connection is dead.
     pub fn send_to(&mut self, peer_id: &AuthorityId, data: D) -> Result<(), SendError> {
-        self.outgoing
+        self.have
             .get(peer_id)
             .ok_or(SendError::PeerNotFound)?
             .unbounded_send(data)
```

Comment:
shouldn't we also remove this connection from `self.have` in case of error?

Reply:
It will be overrriden by any successful connection attempt, until then the only difference removing would make is that the returned errors would be `PeerNotFound` rather than `ConnectionClosed` whenever we attempt to send something. I think the latter is marginally more informative, so I would leave it as is.